stub_uploader/build_wheel.py

```diff
@@ -228,34 +228,40 @@ def collect_package_data(base_path: Path) -> PackageData:
     """
     package_data: dict[str, list[str]] = {}
     for entry in base_path.iterdir():
-        if entry.name == META:
-            # Metadata file entry is added at the end.
+        if is_ignored_distribution_file(entry):
             continue
-        if entry.is_file():
-            if entry.suffix != ".pyi":
-                if entry.suffix not in (".md", ".rst"):
-                    if (
-                        subprocess.run(
-                            ["git", "check-ignore", entry.name], cwd=str(base_path)
-                        ).returncode
-                        != 0
-                    ):
-                        raise ValueError(
-                            f"Only stub files are allowed, not {entry.name!r}"
-                        )
-                continue
-            pkg_name = entry.stem + SUFFIX
+        elif entry.is_file() and entry.suffix == ".pyi":
+            pkg_name = entry.stem
             # Module -> package transformation is done while copying.
-            package_data[pkg_name] = ["__init__.pyi"]
+            stub_files = ["__init__.pyi"]
+        elif entry.is_dir():
+            pkg_name = entry.name
+            stub_files = find_stub_files(str(entry))
         else:
-            if entry.name == TESTS_NAMESPACE:
-                continue
-            pkg_name = entry.name + SUFFIX
-            package_data[pkg_name] = find_stub_files(str(entry))
-        package_data[pkg_name].append(META)
+            raise ValueError(f"Only stub files are allowed, not {entry.name!r}")
+        package_data[pkg_name + SUFFIX] = [*stub_files, META]
```

Comment:
Maybe you won't wanna do that now, but since you're touching this area of code I'll mention it:
With https://github.com/typeshed-internal/stub_uploader/issues/112 (setuptools v69) , the `stub_files` logic could become unnecessary.

According to https://setuptools.pypa.io/en/stable/userguide/miscellaneous.html#controlling-files-in-the-distribution this is an opt-out feature, so you shouldn't need to add anything new either.

Reply:
Let's punt this to a separate PR to keep this as a simple refactoring.